network/p2p/network.go

```diff
@@ -31,18 +31,23 @@ type ReadyDoneAwareNetwork interface {
 // the protocols for handshakes, authentication, gossiping and heartbeats.
 type Network struct {
 	sync.RWMutex
-	logger  zerolog.Logger
-	codec   network.Codec
-	ids     flow.IdentityList
-	me      module.Local
-	mw      network.Middleware
-	top     network.Topology // used to determine fanout connections
-	metrics module.NetworkMetrics
-	rcache  *RcvCache // used to deduplicate incoming messages
-	queue   network.MessageQueue
-	ctx     context.Context
-	cancel  context.CancelFunc
-	subMngr network.SubscriptionManager // used to keep track of subscribed channels
+	logger            zerolog.Logger
+	codec             network.Codec
+	ids               flow.IdentityList
+	me                module.Local
+	mw                network.Middleware
+	top               network.Topology // used to determine fanout connections
+	metrics           module.NetworkMetrics
+	rcache            *RcvCache // used to deduplicate incoming messages
+	queue             network.MessageQueue
+	ctx               context.Context
+	cancel            context.CancelFunc
+	subMngr           network.SubscriptionManager // used to keep track of subscribed channels
+	stateTransition   sync.Mutex
+	ready             chan struct{}
+	done              chan struct{}
+	startupCommenced  bool
+	shutdownCommenced bool
```

Comment:
Please add godoc for this part reflecting why do we need them and what are they used for.